Osx1015 fixes

thehajime opened the pull request:
for track purpose too.

retrage committed:
fix build issue on macOS 10.15

From macOS 10.15, -fstack-check is enabled by default. This option
produces "____chkstk_darwin" undefined symbols[1]. This commit fixes this
issue by passing EXTRA_CFLAGS in build.sh to lkl.

[1] https://forums.developer.apple.com/thread/121887

Signed-off-by: Akira Moroo <retrage01@gmail.com>

retrage committed:
travis: add xcode 11.3 build

Signed-off-by: Akira Moroo <retrage01@gmail.com>

thehajime commented:
https://github.com/ukontainer/frankenlibc/issues/5

10.15 の問題は、xcode11 の問題も含んでいそうでした (10.14+xcode11でも再現しました)。
参考までに。

thehajime commented on rumpkernel/linux.sh:
```diff
@@ -12,11 +12,12 @@ rumpkernel_buildrump()
 	-V RUMP_CURLWP=hypercall -V RUMP_LOCKS_UP=yes \
 	-V MKPIC=no -V RUMP_KERNEL_IS_LIBC=1 \
-	-F CFLAGS=-fno-stack-protector \
+	-F CFLAGS="${EXTRA_CFLAGS}" \
```
これは、`-fno-stack-protector`を残したいですね。

retrage replied:
ローカル環境で試そうとしましたが，`scripts/kconfig/conf  --defconfig Kconfig`でSEGVしてビルドが通らないため憶測ですが，`-fno-stack-protector`を入れてしまうとリンク時に`____chkstk_darwin `が未定義でエラーになったような覚えがあります．`build.sh`で`EXTRA_CFLAGS+=-fno-stack-protector`みたいなことをすればいいはずです．

retrage commented:
macOS  10.15.4でビルドを試したところ，`scripts/kconfig/conf`のSEGVを回避するために以下のpatchが必要なことがわかりました．
```patch
diff --git a/tools/lkl/Makefile.autoconf b/tools/lkl/Makefile.autoconf
index 9ab42b7ce333..2f02a5b96888 100644
--- a/tools/lkl/Makefile.autoconf
+++ b/tools/lkl/Makefile.autoconf
@@ -38,6 +38,7 @@ define macos_host
   KBUILD_CLANG_CFLAGS += -Wno-uninitialized -Wno-sometimes-uninitialized
   KBUILD_CLANG_CFLAGS += -Wno-address-of-packed-member -Wno-unused-value
   KBUILD_CLANG_CFLAGS += -Wno-return-stack-address
+  KBUILD_CLANG_CFLAGS += -fno-stack-check
   export KBUILD_CLANG_CFLAGS
 
   CFLAGS += $$(KBUILD_CLANG_CFLAGS)
```

retrage committed:
lkl: fix build on macOS 10.15

On macOS 10.15, a segmentation fault occurs while running
scripts/kconfig/conf because of stack_not_16_byte_aligned_error[1].
This commit adds a patch that adds -fno-stack-check to host cc flags.

[1] https://forums.developer.apple.com/thread/121887

Signed-off-by: Akira Moroo <retrage01@gmail.com>

retrage committed:
lkl: fix register restoring on fork

As reported on issue #5, fork-test failed after xcode 11. This is
because generated code uses r14 register in expanded RESTORE_REG() in
lkl_restore_register(). The patch adds restored registers to clobber
list in the inline assembly.

Signed-off-by: Akira Moroo <retrage01@gmail.com>

retrage committed:
travis: add python package on xcode 11.3

Signed-off-by: Akira Moroo <retrage01@gmail.com>

thehajime commented:
Fixes around 5e011e37dcceed4921b64e91d3b63bbb8fd990ef commit.
